stage_descriptions/auth-05-uv9.md

```diff
@@ -0,0 +1,85 @@
+In this stage, you'll add support for responding to the `ACL GETUSER` command.
+
+### The `ACL GETUSER` command
+
+The [`ACL GETUSER`](redis.io/docs/latest/commands/acl-getuser/) command is used to list all the ACL rules defined for an existing ACL user.
+
+Example usage:
+
+```bash
+> ACL GETUSER default
+ 1) "flags"
+ 2) 1) "on"
+    2) "nopass"
+    3) "sanitize-payload"
+ 3) "passwords"
+ 4) (empty array)
+ 5) "commands"
+ 6) "+@all"
+ 7) "keys"
+ 8) "~*"
+ 9) "channels"
+10) "&*"
+11) "selectors"
+12) (empty array)
+```
+
+The `ACL GETUSER` command returns a [RESP array](https://redis.io/docs/latest/develop/reference/protocol-spec/#arrays). The elements of the response array are:
+
+1. The literal string `flags`, encoded as a RESP bulk string.
+
+2. List of flags, encoded as a RESP array. Each element of the array is encoded as a RESP bulk string. A user can have multiple flags, so this element is an array.
+
+3. The literal string `passwords`, encoded as a RESP bulk string.
+
+4. List of passwords associated with the user, encoded as a RESP array. Each element is a SHA-256 hash of the original password, encoded as a RESP bulk string. A user can have multiple passwords, so this element is also a RESP array.
+
+5. THe literal string `commands`, encoded as a RESP bulk string.
+
+6. ACL rules for command permissions, encoded as a RESP bulk string.
+
+The official Redis implementation returns other elements as well, which include key permissions, channel permission, and selectors. You can read more about it [here](https://redis.io/docs/latest/commands/acl-getuser/). In our implementation, we'll only focus on user flags, passwords, and command permissions.
+
+- The `off` string in the flags array signifies that the user is not enabled by default. In this stage, you can hardcode this value. We'll get to implementing user flags in the later stages.
+
+- The `-@all` string in the commands signifies that a newly created user has no permissions to execute. In this stage, you can hardcode this value. We'll get to implementing command permissions in the later stages.
+
+### Tests
+
+```bash
+$ ./your_program.sh
+```
+
+It'll then send an `ACL SETUSER` command to create a user, followed by an `ACL GETUSER` command to retrieve the user's information.
+
+```bash
+$ redis-cli
+> ACL SETUSER username
+# Expect: +OK\r\n
+OK
+> ACL GETUSER username
+# Expect RESP array:
+ 1) "flags"
+ 2) 1) "off"
+ 3) "passwords"
+ 4) (empty array)
+ 5) "commands"
+ 6) "-@all"
+```
+
+The tester will validate the following:
```

Comment:
This stage feels a bit overwhelming because of the concepts involved + being asked to hardcode multiple things at once without full knowledge of what they are. 

Can we try an alternate approach? Maybe something like:

- Retrieving user flags
- Updating user flags
- Retrieving user passwords
- Setting user passwords
- Retrieving ...

For each, we'll only test the subset of fields that are relevant for that stage. This way we only introduce one/two concepts per stage and don't have to ask users to hardcode too much